oak_runtime/src/node/http/mod.rs

```diff
@@ -289,6 +303,13 @@ struct HttpRequestHandler {
 
 impl HttpRequestHandler {
     async fn handle(&self, req: Request<Body>) -> Result<Response<Body>, OakStatus> {
+        self.handle_request(req).await.map_err(|err| {
+            warn!("Error when handling the request: {:?}", err);
+            OakStatus::ErrInternal
+        })
+    }
+
+    async fn handle_request(&self, req: Request<Body>) -> Result<Response<Body>, HttpError> {
```

Comment:
Why are `handle` and `handle_request` separate functions?
Could you please add a comment?